apps/system/lockscreen/js/lockscreen_inputpad.js

```diff
@@ -173,6 +189,8 @@
 
   LockScreenInputpad.prototype.handlePassCodeInput =
```

Comment:
As far as I can tell, this function is still only called in response to click events. There is no key handler for the actual keypad, and this will not work on a non-touch device. I don't think it is worth landing any of the PIN stuff until you have it working non-touch.

What's the problem there? Are you having trouble with key events or is this something that you just have not started on yet? If you have questions about this, let me know and I might be able to help.